Add support for the muliple widgets

rwx788 opened the pull request:
Now we have support of all basic widgets.
I've also added optional parameter for the tables, in case unique value is not in the first column, like it's currently.

Required changes:
* [libyui](https://github.com/libyui/libyui/pull/147)
* [libyui-qt](https://github.com/libyui/libyui-qt/pull/117)
* [libyui-qt-rest-api](https://github.com/libyui/libyui-qt-rest-api/pull/2)
* [libyui-ncurses](https://github.com/libyui/libyui-ncurses/pull/82)
* [libyui-ncurses-rest-api](https://github.com/libyui/libyui-ncurses-rest-api/pull/2)

Commit:
Add support to operate on trees with rest-api

Commit:
Bump version to 0.3.0

Commit:
Add possibility to specify column number

Commit:
Update and improve documentation

Commit:
Use boost:split and address other review comments

rwx788 commented:
@lslezak I messed it up and created PR to my original repo and not fork of `libyui-rest-api`

Commit:
Filter on multiple fields and act on single widget only

Commit:
Unite all selection methods and use type to detect action handler

Commit:
Throw error in case item cannot be found in table or tree

Commit:
Add support for MultiLineEdit to set value

Commit:
Add support to set values for YIntField

Commit:
Serialize value for the progress bar

Commit:
Update documentation for select method

Commit:
Add errors when try to select to enter_text for unsupported widget

jknphy commented on README.md:
```diff
@@ -131,25 +127,34 @@ That means anybody on the way could read the user name and the password
+    <li><b>check</b>|<b>uncheck</b>|<b>toggle</b> - check, uncheck or toggle checkbox</li>
+    <li><b>enter_text</b> - set text in the field, requires <b>value</b> parameter</li>
+    <li><b>switch_radio</b> - activate radio button</li>
+    <li><b>select_combo</b> - select value in the combobox, requires <b>value</b> parameter</li>
```
missing to update unique `select` here too.

Commit:
Add support for clicking links in RichText for rest-api

Commit:
Add activate method to support YDumbTab in rest-api

Add support for the YDumbTab. We had to implement activation method, as
in NCurses selection and actual activation are different steps,
similarly to YTree objects.

Also, sort imports alphabetically.

Commit:
Fix typo s/serach/search

Commit:
Use "select" action for radio buttons too

Unifying actions for radio buttons as well

Commit:
Add support for the YSelectionBox

Add support for one more control, worked without extra changes in
libyui.

Commit:
Add support for YItemSelector

Added actions to select, check, uncheck and toggle.

Commit:
Add support for the YMenuButton

The last control in the list is now supported.

Commit:
Modify activation to work for CustomItemSelector

Commit:
Adjust code style as per guidelines

Main change is to put curly bracket on next line for if/else/else if/
conditions.

shundhammer commented on package/libyui-rest-api.changes:
```diff
@@ -1,3 +1,12 @@
+- Add support to operate on many widgets with rest-api
+- Support column index when selecting a row
+- Update documentation
+- Increase SO vesrion to 11
```
Typo: version

shundhammer commented:
LGTM
